src/displayapp/screens/WatchFaceTerminal.cpp

```diff
@@ -0,0 +1,252 @@
+    auto newDateTime = currentDateTime.Get();
+
+    auto dp = date::floor<date::days>(newDateTime);
+    auto time = date::make_time(newDateTime - dp);
+    auto yearMonthDay = date::year_month_day(dp);
+
+    auto year = (int) yearMonthDay.year();
+    auto month = static_cast<Pinetime::Controllers::DateTime::Months>((unsigned) yearMonthDay.month());
+    auto day = (unsigned) yearMonthDay.day();
+    auto dayOfWeek = static_cast<Pinetime::Controllers::DateTime::Days>(date::weekday(yearMonthDay).iso_encoding());
+
+    int hour = time.hours().count();
+    auto minute = time.minutes().count();
+    auto second = time.seconds().count();
+
+    char minutesChar[6];
+    sprintf(minutesChar, "%02d", static_cast<int>(minute));
+
+    char hoursChar[8];
+
+    char ampmChar[3];
+    if (settingsController.GetClockType() == Controllers::Settings::ClockType::H24) {
+      sprintf(hoursChar, "%02d", hour);
+    } else {
+      if (hour == 0 && hour != 12) {
+        hour = 12;
+        sprintf(ampmChar, "AM");
+      } else if (hour == 12 && hour != 0) {
+        hour = 12;
+        sprintf(ampmChar, "PM");
+      } else if (hour < 12 && hour != 0) {
+        sprintf(ampmChar, "AM");
+      } else if (hour > 12 && hour != 0) {
+        hour = hour - 12;
+        sprintf(ampmChar, "PM");
+      }
+      sprintf(hoursChar, "%02d", hour);
+    }
+
+    char secondsChar[5];
+    sprintf(secondsChar, "%02d", static_cast<int>(second));
+
+    auto batteryValue = static_cast<uint8_t>(batteryController.PercentRemaining());
+    char battStr[24];
+    sprintf(battStr, "[BATT]#387b54 %d%\%#", batteryValue);
+    lv_label_set_text(batteryPercent, battStr);
+
+    if (hoursChar[0] != displayedChar[0] || hoursChar[1] != displayedChar[1] || minutesChar[0] != displayedChar[2] ||
+        minutesChar[1] != displayedChar[3] || secondsChar[0] != displayedChar[4] || secondsChar[1] != displayedChar[5]) {
+      displayedChar[0] = hoursChar[0];
+      displayedChar[1] = hoursChar[1];
+      displayedChar[2] = minutesChar[0];
+      displayedChar[3] = minutesChar[1];
+      displayedChar[4] = secondsChar[0];
+      displayedChar[5] = secondsChar[1];
+
+      if (settingsController.GetClockType() == Controllers::Settings::ClockType::H12) {
+        if (hoursChar[0] == '0') {
+          hoursChar[0] = ' ';
+        }
+      }
+
+      char timeStr[42];
+      sprintf(timeStr,
+              "[TIME]#11cc55 %c%c:%c%c:%c%c %s#",
+              hoursChar[0],
+              hoursChar[1],
+              minutesChar[0],
+              minutesChar[1],
+              secondsChar[0],
+              secondsChar[1],
+              ampmChar);
+
+      lv_label_set_text(label_time, timeStr);
+    }
+
+    if ((year != currentYear) || (month != currentMonth) || (dayOfWeek != currentDayOfWeek) || (day != currentDay)) {
+
+      char dateStr[38];
+      sprintf(dateStr, "[DATE]#007fff %04d.%02d.%02d#", short(year), char(month), char(day));
+      lv_label_set_text(label_date, dateStr);
```

Comment:
simplify date string creation

```suggestion
      lv_label_set_text_fmt(label_date, "[DATE]#007fff %04d.%02d.%02d#", short(year), char(month), char(day));
```